web/frontend/src/pages/form/components/FormRow.tsx

```diff
@@ -15,21 +15,25 @@ const FormRow: FC<FormRowProps> = ({ form }) => {
   useEffect(() => {
     if (form.Title === '') return;
     if (isJson(form.Title)) {
-      const ts = JSON.parse(form.Title);
-      setTitles(ts);
+      setTitles(JSON.parse(form.Title));
     } else {
-      const t = { en: form.Title, fr: form.TitleFr, de: form.TitleDe };
-      setTitles(t);
+      setTitles({ en: form.Title, fr: form.TitleFr, de: form.TitleDe });
     }
   }, [form]);
+  const formRowI18n = i18n.createInstance();
+  formRowI18n.init();
+  formRowI18n.changeLanguage(i18n.language);
+  Object.entries(titles).forEach(([lang, title]: any[]) => {
+    if (title) {
+      formRowI18n.addResource(lang, 'form', 'title', title ?? '');
+    }
+  });
```

Comment:
```suggestion
  Object.entries(titles).forEach(([lang, title]: [string, (string | undefined)]) => {
      formRowI18n.addResource(lang, 'form', 'title', title ?? "");
    });
```

However, this will be a bit different: now it even adds an entry for a language with an undefined `title`, but the `title` will be empty. If you want to avoid adding empty titles, replace the callback with this:

```
    if (title !== undefined){
      addResource(lang, 'form', 'title', title);
    }
```

Reply:
If that doesn't work, then you need to add a type definition to `titles`:

```
titles: {[key: string]: (string | undefined)}
```

Play around here:

https://www.typescriptlang.org/play?ssl=3&ssc=1&pln=4&pc=1#code/MYewdgzgLgBFCWUA2BTCAuGBvA2gaxQE9NoAneMAcwF1MAKMiymAHxgFcwATFAMwpRcAlAF8YAXmwAiAIZTMUgEZSANDCXyO3PgK4iA3AChDvTsATgYMrlwBKaEO1LAUdGSSjkqaxR6+U1YD8mNS5gqiEsQxgYmFBIEFQAOiQQSjdQoSMRYwB5RQArFHMklDBPeDQ6BGQ0ISTeEFIAURlgAAs6OhwkGW84RFRaGBxGfoYKqlYtHn4wQSFqIQkAPmxo2PheGGrBlBgAQnFJTlndSI3Yqxt7CEdnV17+gHJG0gBbZ7VnmtQvgdqWUuMByMREQKAA

Reply:
Yes, it would also be true. So you would need

```
if (title !== undefined && title !== "")
```

Reply:
Or you could also use

```
if (str?.length > 0)
```

Reply:
Just for posteriority: sorry for the noise...

1. `(str !== undefined && str.length > 0)` is the most correct, but it's long
2. `(str?.length > 0)` potentially compares `undefined` and `0`, which is wrong and typescript will complain in strict mode - because I didn't read [StackOverflow](https://stackoverflow.com/a/3261380/1573347) correctly
3. `(str?.length)` would be better - now we're coercing either an `undefined` or a `number` into a `bool`
4. `(str)` coerces a `string` or an `undefined` into a boolean, which works, but shouldn't :)

I think I would go with 3., but 4. is also OK.

My preferred way would be:

```
Objects.entries(titles)
  .filter([_, title] => title?.length)
  .forEach([lang, title] => ...
```

But that's also wrong.